doc/ansible/user-guide.md

```diff
@@ -0,0 +1,279 @@
+
+Now that we have the spec defined, we can define what Ansible is actually executed on resource changes. Since this is an Ansible Role, the default behavior will be to execute the tasks in `roles/Memcached/tasks/main.yml`. We want Ansible to create a deployment if it does not exist which runs the `memcached:1.4.36-alpine` image. Ansible 2.5+ supports the [k8s Ansible Module](https://docs.ansible.com/ansible/2.6/modules/k8s_module.html) which we will leverage to control the deployment definition.
+
+Modify `roles/Memcached/tasks/main.yml` to look like the following:
+```yaml
+---
+- name: start memcached
+  k8s:
+    definition:
+      kind: Deployment
+      apiVersion: apps/v1
+      metadata:
+        name: '{{ meta.name }}-memcached'
+        namespace: '{{ meta.namespace }}'
+      spec:
+        replicas: {{size}}
+        selector:
+          matchLabels:
+            app: memcached
+        template:
+          metadata:
+            labels:
+              app: memcached
+          spec:
+            containers:
+            - name: memcached
+              command:
+              - memcached
+              - -m=64
+              - -o
+              - modern
+              - -v
+              image: "memcached:1.4.36-alpine"
+              ports:
+                - containerPort: 11211
+
+```
+
+It is important to note that we used the `size` variable to control how many replicas of the Memcached deployment we want. We set the default to `1` but any user can create a Custom Resource which overwrites the default.
+
+### Build and run the operator
+
+Before running the operator, Kubernetes needs to know about the new custom resource definition the operator will be watching.
+
+Deploy the CRD:
+
+```sh
+$ kubectl create -f deploy/crd.yaml
+```
+
+Once this is done, there are two ways to run the operator:
+
+- As pod inside Kubernetes cluster
+- As go program outside cluster using `operator-sdk`
+
+#### 1. Run as pod inside a Kubernetes cluster
+
+Run as pod inside a Kubernetes cluster is preferred for production use.
+
+Build the memcached-operator image and push it to a registry:
+```
+$ operator-sdk build quay.io/example/memcached-operator:v0.0.1
+$ sed -i 's|REPLACE_IMAGE|quay.io/example/memcached-operator:v0.0.1|g' deploy/operator.yaml
+$ docker push quay.io/example/memcached-operator:v0.0.1
+```
+
+Kubernetes deployment manifests are generated in `deploy/operator.yaml`. The deployment image is set to the container image specified above.
+
+Deploy the memcached-operator:
+
+```sh
+$ kubectl create -f deploy/rbac.yaml
+$ kubectl create -f deploy/operator.yaml
+```
+
+Verify that the memcached-operator is up and running:
+
+```sh
+$ kubectl get deployment
+NAME                     DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+memcached-operator       1         1         1            1           1m
+```
+
+#### 2. Run outside the cluster
+
+This method is preferred during development cycle to deploy and test faster.
```

Comment:
This method is preferred during the development cycle to speed up deployment and testing.